server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -533,7 +583,8 @@ public void writeTo(StreamOutput out) throws IOException {
             args[5] != null && (boolean) args[5],
             args[6] != null && (boolean) args[6],
             args[7] != null && (boolean) args[7],
-            args[8] != null && (boolean) args[8]
+            args[8] != null && (boolean) args[8],
+            args[9] != null ? IndexMode.valueOf(((String) args[9]).toUpperCase(Locale.ROOT)) : null
```

Comment:
Same here for using a parsing function rather than the `valueOf`, so we have a better error message